src/phoenix/server/api/types/DatasetExample.py

```diff
@@ -1,13 +1,37 @@
 from datetime import datetime
+from typing import Optional
 
 import strawberry
-from strawberry.relay.types import Node, NodeID
+from strawberry import UNSET
+from strawberry.relay.types import GlobalID, Node, NodeID
+from strawberry.types import Info
 
+from phoenix.server.api.context import Context
 from phoenix.server.api.types.DatasetExampleRevision import DatasetExampleRevision
+from phoenix.server.api.types.DatasetVersion import DatasetVersion
+from phoenix.server.api.types.node import from_global_id_with_expected_type
 
 
 @strawberry.type
 class DatasetExample(Node):
     id_attr: NodeID[int]
     created_at: datetime
-    revision: DatasetExampleRevision
+    version_id: strawberry.Private[Optional[int]] = None
+
+    @strawberry.field
+    async def revision(
+        self,
+        info: Info[Context, None],
+        dataset_version_id: Optional[GlobalID] = UNSET,
+    ) -> DatasetExampleRevision:
```

Comment:
maybe should be optional in case the `dataset_version_id` is one where the example is deleted?
```suggestion
        dataset_version_id: Optional[GlobalID] = UNSET,
    ) -> Optional[DatasetExampleRevision]:
```

Reply:
We'll need to do that if we need to support querying for deleted examples. There are additional changes that need to be made in the `examples` connection and node interface to support that.

Reply:
Punting on this one for now, will revisit if necessary.